update-versions.py

```diff
@@ -0,0 +1,145 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Prepare new artman release.
+
+This script updates SHA hashes of latest commits for both gapic-generator
+and googleapis, and updates version number for subsequent (auto-)release.
+"""
+
+import argparse
+import os.path
+import re
+import requests
+import sys
+from ruamel.yaml import YAML
+
+dockerfile_googleapis_regex = r'^(ENV GOOGLEAPIS_HASH )\S+()'
+dockerfile_gapic_generator_regex = r'^(ENV GAPIC_GENERATOR_HASH )\S+()'
+dockerfile_version_regex = r'^(ENV ARTMAN_VERSION )\S+()'
+setup_py_version_regex = r"^(current_version = ')\S+(')"
+
+github_gapic_generator_project = 'googleapis/gapic-generator'
+github_googleapis_project = 'googleapis/googleapis'
+
+
+def get_latest_commit_sha(project):
+    url = 'https://api.github.com/repos/%s/branches/master' % project
+    # try to get GitHub credentials from config
+    auth = None
+    artman_config = os.path.join(os.environ['HOME'], '.artman', 'config.yaml')
+    try:
+        yaml = YAML()
+        with open(artman_config) as config_yaml:
+            config = yaml.load(config_yaml)
+            github_username = config['github']['username']
+            github_token = config['github']['token']
+            auth = requests.auth.HTTPBasicAuth(github_username, github_token)
+    except:
+        print('Cannot get GitHub token from %s, trying anonymously...' %
+              artman_config)
+
+    sha = None
+    try:
+        response = requests.get(url, auth=auth)
+        repo = response.json()
+        sha = repo['commit']['sha']
+    except:
+        print('Cannot get latest commit SHA. GitHub rate limit?')
+
+    return sha
+
+
+def update_file(filename, regex, version):
+    lines = []
+    with open(filename) as f:
+        for line in f:
+            line = re.sub(regex, r'\g<1>' + version + r'\g<2>', line)
+            lines.append(line)
+    with open(filename, 'w') as f:
+        f.write(''.join(lines))
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Prepare new artman release.')
+    parser.add_argument(
+        'version', nargs='?', help='Set version (MAJOR.MINOR.PATCH).')
+    parser.add_argument(
+        '--fetch-googleapis',
+        action='store_true',
+        help=
+        'Fetch the latest googleapis commit SHA from GitHub and store in Dockerfile.'
+    )
+    parser.add_argument(
+        '--fetch-gapic-generator',
+        action='store_true',
+        help=
+        'Fetch the latest gapic-generator commit SHA from GitHub and store in Dockerfile.'
+    )
+
+    args = parser.parse_args()
+
+    if not args.version and not args.fetch_googleapis and not args.fetch_gapic_generator:
+        parser.print_help()
+        return 1
```

Comment:
You aren't calling `exit()` using the return value of `main()` any more so this will currently have no effect.

I have personally just thrown `ValueError` in my own scripts which results in a non-zero exit code automatically. I'm not sure what the best practice is - @lukesneeringer do you have an opinion?